Packet++/src/SingleCommandTextProtocol.cpp

```diff
@@ -5,25 +5,31 @@
 
 #define ASCII_HYPHEN 0x2d
 #define ASCII_SPACE 0x20
+#define MAX_COMMAND_LENGTH 9 // From SMTP command "STARTTLS" + 1 byte hyphen or space
+#define MIN_PACKET_LENGTH 2 // CRLF
 
 namespace pcpp
 {
 
 	size_t SingleCommandTextProtocol::getArgumentFieldOffset() const
 	{
 		size_t maxLen;
-		if (m_DataLen < 5)
+		if (m_DataLen < MAX_COMMAND_LENGTH)
 			maxLen = m_DataLen;
 		else
-			maxLen = 5;
+			maxLen = MAX_COMMAND_LENGTH;
 
-		// Find <SP> if exists
-		uint8_t *pos = (uint8_t *)memchr(m_Data, ASCII_SPACE, maxLen);
+		// To correctly detect multi-line packets with the option containing a space in
+		// the first MAX_CONTENT_LENGTH bytes, search the hyphen first otherwise it thinks
+		// the space in option as a command delimiter
+
+		// Find Hyphen "-" if exists
+		uint8_t *pos = (uint8_t *)memchr(m_Data, ASCII_HYPHEN, maxLen);
 		if (pos)
 			return pos - m_Data;
 
-		// Find Hyphen "-" if exists
-		pos = (uint8_t *)memchr(m_Data, ASCII_HYPHEN, maxLen);
+		// Find <SP> if exists
+		pos = (uint8_t *)memchr(m_Data, ASCII_SPACE, maxLen);
```

Comment:
Maybe we should check what comes sooner - `<SP>` or `-`?
Since we look at the first 9 bytes of the string there could theoretically be a case like this:
`CMD a-b-c`

(`CMD` is some command in some protocol)

In this case, the result will be incorrect because it'll consider the `-` as separator instead of `<SP>`